src/dog.sol

```diff
@@ -173,8 +173,15 @@ contract Dog {
 
             dart = min(art, mul(room, WAD) / rate / milk.chop);
 
+            uint256 mart = mul(art - dart, rate);
+
+            if (mart < dust) {
+                // avoid leaving a dusty vault to prevent unliquidatable vaults
+                dart = add(dart, mart);
+            }
+
             // Verify that CDP is not left in a dusty state
-            require(dart == art || mul(art - dart, rate) >= dust, "Dog/leaves-dust");
+            require(dart == art || mart >= dust, "Dog/leaves-dust");
```

Comment:
Actually this `require` is not needed anymore, as if `mart` is lower than `dust` will be taken care in the prev step. So I think the whole thing can be refactor to this:

Reply:
```
if (mul(art - dart, rate) < dust) {
    // avoid leaving a dusty vault to prevent unliquidatable vaults
    dart = art;
}
```
(removing the `require` and `uint256 mart = mul(art - dart, rate);`)